packages/opentelemetry-sdk-trace-node/src/NodeTracerProvider.ts

```diff
@@ -68,10 +70,22 @@ export class NodeTracerProvider extends BasicTracerProvider {
     super.register(config);
   }
 
-  protected override  _getPropagator(name: string): TextMapPropagator | undefined {
+  /**
+   * TS cannot yet infer the type of this.constructor:
+   * https://github.com/Microsoft/TypeScript/issues/3841#issuecomment-337560146
+   * Do not override either of the getters in your child class unless you really need to.
+   * We have to do it here because the previous version of BasicTracerProvider's getters
+   * had BasicTracerProvider's static method hardcoded and we want do undo that behavior.
+   */
+  protected override _getPropagator(name: string): TextMapPropagator | undefined {
     return (
-      super._getPropagator(name) ||
-      NodeTracerProvider._registeredPropagators.get(name)?.()
-    );
+      (this.constructor as typeof BasicTracerProvider)._registeredPropagators
+    ).get(name)?.();
+  }
+
+  protected override _getSpanExporter(name: string): SpanExporter | undefined {
+    return (
+      (this.constructor as typeof BasicTracerProvider)._registeredExporters
+    ).get(name)?.();
   }
```

Comment:
Now that the base class implements this logic, do we still need to override these functions in child class? 
I understood that it's enough to add the values to the override `_registeredPropagators` to make the base class pick it up automatically.

Reply:
I see there is already a comment about it above.
Since versions are pinned, I too think it's safe to remove them